src/session.rs

```diff
@@ -64,7 +65,11 @@ impl Session {
                 break;
             }
             // Print as we decode.
-            print!("{} ", self.colorhost);
+            // Without the lock, when multiple commands output to stdout,
+            // lines from different commands get mixed.
+            let stdout = std::io::stdout();
+            let mut lock = stdout.lock();
```

Comment:
I think using `guard` as the variable name is better.

Reply:
Done!